### DISCUSSION TITLE
Using tags in Blade view

### DISCUSSION THREAD

**Discussion by ajcsilva** [duncanmcclean/simple-commerce#790] [Help] [answered]
Hi all,

I'm running into some trouble using tags in my blade views, specifically `sc:cart:addItem.` 

I'm using the tag like this: `{!! Statamic::tag('sc:cart:addItem')->param('product', $id)->fetch() !!}` and it returns an Array to string conversion error.

I tracked down the issue to this parse function in the Statamic codebase:

```php
    // statamic/cms/src/Tags/Tags.php
    /**
     * Parse the tag pair contents.
     *
     * @param  array  $data  Data to be parsed into template
     * @return string
     */
    public function parse($data = [])
    {
        if ($scope = $this->params->get('scope')) {
            $data = Arr::addScope($data, $scope);
        }

        if (! $this->parser) {
            return $data;
        }

        return Antlers::usingParser($this->parser, function ($antlers) use ($data) {
            return $antlers
                ->parse($this->content, array_merge($this->context->all(), $data))
                ->withoutExtractions();
        });
    }
```
It seems that `$this->parser` is null and is returning an array when it should be returning a string. 

I'm stuck using Blade for the time being so any help here is much appreciated. Thank you!

**Comment by duncanmcclean** [comment, marked as answer]
Hey!

I don't have much experience with Blade & Statamic but from what I can tell from [the documentation](https://statamic.dev/blade) & from [a brief conversion on Discord](https://discord.com/channels/489818810157891584/489819906540568593/1067938171347816489) is that you can't use any of Simple Commerce's "form tags" in Blade because they need to have 'content' inside them (for example: the HTML that goes in the middle of the start & end tags)

```antlers
{{ sc:cart:addItem }}
  <input type="hidden" name="product" value="{{ id }}">
  <input type="number" name="quantity" value="2">
{{ /sc:cart:addItem }}
```

I've commented on an open Statamic feature request, statamic/ideas#802 as hopefully it's something that can be added in future.

Anyway, as a workaround, I think you have two options....

**1. Use Antlers for the 'add to cart form'**

You don't have to stick with one templating language for your entire site, you can mix and match. 

So, on your product page where you need to add the 'add to cart form', `@include` an Antlers file (the same way you would for a Blade partial, just use the `.antlers.html` file extension). Make sure to pass in any variables you need to the partial, like the product ID.

Then in the Antlers file, you can use Simple Commerce's "form tags".

**2. Create the `<form>`s yourself**

Simple Commerce's "form" tags essentially wrap your HTML from inside the tag inside a `<form>` tag with a set action/method and some hidden parameters.

However, if you need to, you can hard code the code for these forms yourself.

For example, my `sc:cart:addItem` example might produce this HTML output:

```html
<form method="POST" action="http://sc-sandbox.test/!/simple-commerce/cart-items">
    <input type="hidden" name="_token" value="the_csrf_token">
    <input type="hidden" name="_redirect" value="a_big_encrypted_string" /> <!-- This is optional. With the setting I talk about below set to true, you can make the value of this a relative URL, like "/cart" -->
    <input type="hidden" name="_error_redirect" value="a_big_encrypted_string" /> <!-- This is optional. With the setting I talk about below set to true, you can make the value of this a relative URL, like "/cart" -->
    <input type="hidden" name="_request" value="a_big_encrypted_string" />  <!-- This is optional. With the setting I talk about below set to true, you can make the value of this the name of a Form Request in your App\Http\Requests folder (for validation) -->

    <input type="hidden" name="product" value="product-id">
    <input type="number" name="quantity" value="2">
</form>
```

You can find out what each of the different "form tags" produce by using the Antlers Tags somewhere on the site (or in a blank site using the starter kit), then adjusting it for your case.

Also... if you go down this route, you will need to set `disable_form_parameter_validation` to `true` in your `config/simple-commerce.php` config file. Usually SC encrypts the values of the `_redirect`, `_error_redirect`, `_request` parameters for security but it won't be able to do that if you're not using the provided tags. 

---

Hopefully that all made sense - sorry for the bad news & that I just threw a wall of text at you. Hopefully some of that will lead you to find a solution!

Let me know if you have any further questions!

**Comment by ajcsilva** [comment]
Ahh including antlers templates from blade will work just fine. Thanks so much for point that out. 

Thanks for taking the time on on this and for all the work you do! @duncanmcclean

**Reply by duncanmcclean** [reply]
Awesome!

No problem at all - have a great rest of your day! 😄

**Comment by duncanmcclean** [comment]
Proper Blade support for Simple Commerce is in the pipeline - see #792 👀

**Reply by duncanmcclean** [reply]
And... it's just been tagged. If you want to take a look, it's [covered in the docs](https://simple-commerce.duncanmcclean.com/tags#content-using-the-form-tags-in-blade).
